### PR TITLE
Add Loading page for React Native authenticator

### PR DESCRIPTION
**Pull request by rlmartin**
This is necessary because Auth.currentAuthenticatedUser() returns a Promise.
When a user is logged in and the app is reloaded, there is a brief flicker
during which the SignIn screen is shown before the authenticated app
appears. This Loading screen replaces that flicker and will toggle to
either signIn or signedIn, as appropriate.

### CODE REVIEW

**Comment by powerful23** [comment]
@rlmartin hey we are going to merge #199 into library sooner or later so could you update this pr to fit in that merge. Thanks.

**Comment by rlmartin** [comment]
@powerful23 done - thanks.

File: packages/aws-amplify-react-native/src/Auth/Loading.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+import React from 'react';
+import { View, Text } from 'react-native';
+import { I18n } from 'aws-amplify';
+import AuthPiece from './AuthPiece';
+import { Header } from '../AmplifyUI';
+
+export default class Loading extends AuthPiece {
+    render() {
```
Comment (powerful23) [inline packages/aws-amplify-react-native/src/Auth/Loading.js]: Change ```render``` to ```showComponent```

File: packages/aws-amplify-react-native/src/Auth/Loading.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * and limitations under the License.
+ */
+
+import React from 'react';
+import { View, Text } from 'react-native';
+import { I18n } from 'aws-amplify';
+import AuthPiece from './AuthPiece';
+import { Header } from '../AmplifyUI';
+
+export default class Loading extends AuthPiece {
```
Comment (powerful23) [inline packages/aws-amplify-react-native/src/Auth/Loading.js]: add ```_validAuthStates```

**Comment by powerful23** [comment]
@rlmartin sorry I didn't make it clear. Since now the AuthPiece has been changed, could you replace that ```render``` with ```showComponent``` and also add the ```_validAuthStates```? Thanks!

**Comment by rlmartin** [comment]
Thanks, good catch. I had a separate commit on a local branch in which I brought all these commits together. I'll add it to this PR when I get home tonight.

**Comment by mlabieniec** [comment]
@rlmartin just waiting on the few small changes here and we can get this merged. Thanks!

**Comment by rlmartin** [comment]
@mlabieniec yup, sorry - slipped my mind last night. Here is the additional commit.

**Comment by pulpfree** [comment]
Curious if anyone knows if this change is planned for the `aws-amplify-react` package?

**Comment by Legym** [comment]
I have been banging my head for hours on this issue. I cannot seem to find any other issues pertaining to this issue!

**Pull request merged**